samples/custom_dispatch/vulkan/shaders/udon_spec.mlir

```diff
@@ -0,0 +1,76 @@
+// The configuration used for executable compilation.
+// This specifies the device configurations that support this custom kernel.
+#spirv_target = #hal.executable.target<"vulkan", "vulkan-spirv-fb", {
+  spirv.target_env = #spirv.target_env<
+    #spirv.vce<v1.3, [Shader, GroupNonUniform, GroupNonUniformBallot], [SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers]>,
+    #spirv.resource_limits<max_compute_workgroup_size = [128, 128, 64], subgroup_size = 64>
+  >
+}>
+
+module attributes {transform.with_named_sequence} {
+  func.func @argmax_1d_f32_entry_point(%arg0: tensor<1x?xf32>) -> tensor<1xi64> {
+    %c1 = arith.constant 1 : index
+    %dim = tensor.dim %arg0, %c1 : tensor<1x?xf32>
+    // Note: This is not safe if the dim size exceeds INT32_MAX. To pass a 64
+    // bit value it must be broken down into two 32-bit values for the high and
+    // low bits.
+    %dim_i32 = arith.index_cast %dim : index to i32
+    // Inline external dispatch that conforms to the ABI that the kernel
+    // requires. This is the primary reason for the surrounding function as
+    // details like tensor shape and push constants need to line up after
+    // splicing in the custom dispatch. This allows the kernel author to manage
+    // such details by hand without needing the rewrite patterns to worry about
+    // things like order of push constants.
+    %4 = hal.dispatch.extern "argmax_1d_f32"[%dim](%dim_i32, %arg0) : (i32, tensor<1x?xf32>) -> tensor<1xi64>
+      count(%device: !hal.device, %workload: index) -> (index, index, index) {
+        %c1_0 = arith.constant 1 : index
+        hal.return %c1_0, %c1_0, %c1_0 : index, index, index
+      }   
+      layout(#hal.pipeline.layout<push_constants = 1, sets = [
+        <0, bindings = [
+            <0, storage_buffer, ReadOnly>,
+            <1, storage_buffer>
+        ]>
+      ]>)
+      bindings([
+        #hal.interface.binding<0, 0>, 
+        #hal.interface.binding<0, 1>
+      ])  
+      objects({
+        #spirv_target ordinal(0) = [ 
+          #hal.executable.object<{
+            path = "samples/custom_dispatch/vulkan/shaders/one_workgroup_argmax_subgroup_f32.spv"
+          }>
+        ]
+      })
+    return %4 : tensor<1xi64>
+  }
+
+  transform.named_sequence @match_argmax(%current: !transform.any_op {transform.readonly}) -> (!transform.any_op) {
+    transform.yield %current : !transform.any_op
+  }
+
+  transform.named_sequence @cast_and_call_argmax(%argmax: !transform.any_op {transform.readonly}) {
```

Comment:
I am not entirely clear on how the argmax matching is happening.

Reply:
Still adding that part